### PR TITLE
Add Storage Module to POM

### LINKED ISSUE #4316: Turn on Linting for Blobs
Turn on CheckStyle, SpotBugs, and JavaDoc linting for Azure Storage Blobs

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #4316 

- Turns on linting for Azure Storage Blobs
- Adds Storage as a module to pom.client.xml
- Updates Azure Core versions and updates the depending libraries to the correct versions

### CODE REVIEW

File: eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml [hunk 1]
```diff
@@ -107,6 +107,10 @@
     <Class name="~com\.microsoft\.azure\.(.+)Impl"/>
     <Bug pattern="NP_LOAD_OF_KNOWN_NULL_VALUE"/>
   </Match>
+  <Match>
+    <Class name="~com\.azure\.(.+)Impl"/>
```
Comment (JonathanGiles) [inline eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: Use the `<Or>` pattern you can see elsewhere to specify multiple classes for a particular bug pattern

File: eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml [hunk 2]
```diff
@@ -360,6 +367,10 @@
     <Package name="com.microsoft.azure.storage.blob"/>
     <Bug pattern="NM_CONFUSING"/>
   </Match>
+  <Match>
+    <Class name="com.azure.storage.blob.HTTPGetterInfo"/>
+    <Bug pattern="NM_CONFUSING"/>
+  </Match>
```
Comment (JonathanGiles) [inline eng/code-quality-reports/src/main/resources/spotbugs/spotbugs-exclude.xml]: Same 'Or' comment here

File: pom.client.xml [hunk 3]
```diff
@@ -385,6 +385,10 @@
                 <title>Azure Key Vault</title>
                 <packages>com.azure.security.keyvault*</packages>
               </group>
+              <group>
+                <title>Azure Storage</title>
+                <packages>com.azure.storage*</packages>
+              </group>
```
Comment (JonathanGiles) [inline pom.client.xml]: Should this be split up into blob, files and queues?

File: storage/client/blob/src/test/java/com/azure/storage/blob/LargeFileTest.java [hunk 4]
```diff
@@ -9,7 +12,7 @@
 
 public class LargeFileTest {
     private static final Random RANDOM = new Random();
-    private static final String filePath = "C:\\Users\\jianghlu\\10g.dat";
```
Comment (sima-zhu) [inline storage/client/blob/src/test/java/com/azure/storage/blob/LargeFileTest.java]: Really weird to see person's name here

File: storage/client/blob/src/samples/java/blob/StorageErrorHandlingExample.java [hunk 5]
```diff
@@ -18,42 +19,42 @@
 public class StorageErrorHandlingExample {
 
     public static void main(String[] args) {
-        ContainerClient containerClient = ContainerClient.containerClientBuilder().endpoint("https://account.blob.core.windows.net/mycontainer")
+        ContainerClient containerClient = new ContainerClientBuilder().endpoint("https://account.blob.core.windows.net/mycontainer")
```
Comment (sima-zhu) [inline storage/client/blob/src/samples/java/blob/StorageErrorHandlingExample.java]: Any reason to change like this way?

Comment (alzimmermsft) [reply storage/client/blob/src/samples/java/blob/StorageErrorHandlingExample.java]: The builder static on the client was removed based on #3871

File: storage/client/blob/src/test/java/com/azure/storage/blob/LargeFileTest.java (hunk 4 above)
Comment (alzimmermsft) [reply storage/client/blob/src/test/java/com/azure/storage/blob/LargeFileTest.java]: I noticed that too, looks like some testing unit tests made their way into the central repo, shouldn't be an issue just looks weird.

File: storage/client/blob/src/main/java/com/azure/storage/blob/BlockBlobClient.java [hunk 6]
```diff
@@ -161,32 +163,42 @@ public BlobOutputStream getBlobOutputStream(BlobAccessConditions accessCondition
...
             .upload(fbb.subscribeOn(Schedulers.elastic()), length, headers, metadata, accessConditions)
-            .map(rb -> new SimpleResponse<>(rb, new BlockBlobItem(rb.deserializedHeaders())));;
+            .map(rb -> new SimpleResponse<>(rb, new BlockBlobItem(rb.deserializedHeaders())));
 
         try {
             return Utility.blockWithOptionalTimeout(upload, timeout);
-        }
-        catch (UncheckedIOException e) {
+        } catch (UncheckedIOException e) {
             throw e.getCause();
```
Comment (sima-zhu) [inline storage/client/blob/src/main/java/com/azure/storage/blob/BlockBlobClient.java]: I have seen you have the Utility.blockWithOptionalTimeout for simplifying try/catch somewhere else. Why you try/catch again here? Can you move the checking into the utility methods?

Comment (alzimmermsft) [reply storage/client/blob/src/main/java/com/azure/storage/blob/BlockBlobClient.java]: The blockWithOptionalTimeout simplifies and if/else block not a try/catch block. It shouldn't be used for catching IOExceptions everywhere

**Pull request merged**